components/sponsor/class-sponsor.php

```diff
@@ -0,0 +1,274 @@
+			'theme' => 'module',
+		];
+	}
+
+	/**
+	 * Hook into post being set.
+	 *
+	 * @return self
+	 */
+	public function post_has_set() : self {
+
+		$label = (string) get_post_meta( $this->get_post_id(), 'label', true );
+		if ( empty( $label ) ) {
+			$label = self::get_default_eyebrow_label();
+		}
+		$this->set_config( 'label', $label );
+		$this->set_config( 'link', get_post_meta( $this->get_post_id(), 'link', true ) );
+
+		// Append the message and short_message as HTML components.
+		$this->append_children(
+			[
+				// Message HTML component.
+				( new \WP_Components\HTML() )
+					->set_config( 'context', 'message' )
+					->set_config( 'content', (string) get_post_meta( $this->get_post_id(), 'message', true ) ),
+				self::get_content_below_sponsor_component(),
+			]
+		);
+
+		// Add thes sponsor logo as a child component.
+		$logo_id = get_post_meta( $this->get_post_id(), 'logo_id', true );
+		if ( ! empty( $logo_id ) ) {
+			$this->append_child(
+				// Note: We're mixing component libraries here. Sponsor uses WP
+				// Components, and this image class is an older version. This
+				// is because to use the WP Component's image we'd have to
+				// refactor quite a bit, so we'll come back to this later.
+				\Civil_First_Fleet\Component\image()
+					->set_post_id( $logo_id )
+					->size( 'sponsor-logo' )
+					->aspect_ratio( false )
+					->set_data( 'alt', $this->wp_post_get_title() )
+					->disable_lazyload()
+			);
+		}
+
+		return $this;
+	}
+
+	/**
+	 * Get the default eyebrow label text.
+	 *
+	 * @return string
+	 */
+	public static function get_default_eyebrow_label() {
+		return __( 'Supported Today By', 'civil-first-fleet' );
+	}
+
+	/**
+	 * Get the FM fields used for the sponsor post type.
+	 *
+	 * @return array
+	 */
+	public static function get_fm_fields() {
+		return [
+			'label'   => new \Fieldmanager_Textfield(
+				[
+					'label'         => __( 'Eyebrow Label', 'civil-first-fleet' ),
+					'default_value' => self::get_default_eyebrow_label(),
+				]
+			),
+			'link'    => new \Fieldmanager_Link( __( 'Link', 'civil-first-fleet' ) ),
+			'logo_id' => new \Fieldmanager_Media( __( 'Logo', 'civil-first-fleet' ) ),
+			'message' => new \Fieldmanager_RichTextArea( __( 'Message', 'civil-first-fleet' ) ),
+		];
+	}
+
+	/**
+	 * Get the FM fields used to schedule sponsors.
+	 *
+	 * @return array
+	 */
+	public static function get_schedule_fm_fields() : array {
+		return [
+			'schedules' => new \Fieldmanager_Group(
+				[
+					'label'          => __( 'Sponsor Schedules', 'civil-first-fleet' ),
+					'limit'          => 0,
+					'extra_elements' => 0,
+					'label'          => __( 'New Schedule', 'civil-first-fleet' ),
+					'label_macro'    => [ '%s', 'sponsor_id' ],
+					'add_more_label' => __( 'Add Schedule', 'civil-first-fleet' ),
+					'collapsed'      => true,
+					'collapsible'    => true,
+					'sortable'       => true,
+					'children'       => [
+						'sponsor_id'      => new \Fieldmanager_Select(
```

Comment:
I think an autocomplete would be better here in case a site ends up with more than 10 sponsors